python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,1089 @@
+        #     see if it actually was clipped.
+        p_Im = ImageF(bbox)
+        tmpSubim = ImageF(p_Im, psf_box, PARENT)
+        tmpSubim += psf_img
+
+        return p_Im
+
+    @staticmethod
+    def genDipoleModel(x, flux, xcenPos, ycenPos, xcenNeg, ycenNeg, fluxNeg=None,
+                       b=None, x1=None, y1=None, xy=None, x2=None, y2=None,
+                       bNeg=None, x1Neg=None, y1Neg=None, xyNeg=None, x2Neg=None, y2Neg=None,
+                       debug=False, **kwargs):
+        """
+        Generate dipole model with given parameters.
+        This is the functor whose sum-of-squared difference from data is minimized by `lmfit`.
+
+        Parameters
+        ----------
+        x : numpy.array
+           Input independent variable. Used here as the grid on which to compute the background
+           gradient model.
+        flux : float
+           Desired flux of the positive lobe of the dipole
+        xcenPos : float
+           Desired x-centroid of the positive lobe of the dipole
+        ycenPos : float
+           Desired y-centroid of the positive lobe of the dipole
+        xcenNeg : float
+           Desired x-centroid of the negative lobe of the dipole
+        ycenNeg : float
+           Desired y-centroid of the negative lobe of the dipole
+        fluxNeg : float, optional
+           Desired flux of the negative lobe of the dipole, set to 'flux' if None
+        b, x1, y1, xy, x2, y2 : float, optional
+           Gradient parameters for positive lobe, see `genBgGradientModel`.
+        bNeg, x1Neg, y1Neg, xyNeg, x2Neg, y2Neg : float, optional
+           Gradient parameters for negative lobe, see `genBgGradientModel`. Set to the corresponding
+           positive values if None.
+        **kwargs :
+           Keyword arguments passed through `lmfit` and used by this functor:
+           {
+              psf : afw.detection.Psf
+                 Psf model used to generate the 'star'
+              rel_weight : float
+                 Used to signify if positive/negative images are to be included (!= 0. if yes)
+              bbox : afw.geom.BoundingBox
+                 Bounding box containing region to be modelled
+           }
+
+        Returns
+        -------
+        numpy.array of width/height matching input bbox, containing dipole model with
+        given centroids and flux(es). If rel_weight = 0., this is a 2-d array with dimensions
+        matching those of bbox; otherwise a stack of three such arrays, representing the dipole
+        (diffim), positive and negative images respectively.
+        """
+
+        psf = kwargs.get('psf')
+        rel_weight = kwargs.get('rel_weight')  # if > 0, we're including pre-sub. images
+        fp = kwargs.get('footprint')
+        bbox = fp.getBBox()
+
+        if fluxNeg is None:
+            fluxNeg = flux
+
+        if debug:
+            print('%.2f %.2f %.2f %.2f %.2f %.2f' % (flux, fluxNeg, xcenPos, ycenPos, xcenNeg, ycenNeg))
```

Comment:
I cannot figure out how to catch logging ouput in an ipython notebook. Do you know of a way?